authz.go

```diff
@@ -184,9 +185,16 @@ func (a *authorizer) authorize(ctx context.Context, sr authz.SubjectAccessReview
 	var via string
 	var client *client
 	if a.AuthorizationConfig.Config.UseCache {
-		decision := a.useCacheEval(log, principal, checks)
-		if decision != nil && !a.AuthorizationConfig.Config.DryRun {
-			return decision
+		// check syncer's last contact time with athenz, if it is more than two hours,
+		// fall back to zms/zts.
+		if a.AuthorizationConfig.Config.Cache.cacheStatus {
```

Comment:
This cache status check needs to have a read lock around it. Maybe we can move this check inside the authorize function in zpeclient as we're locking there?